### PR TITLE
Changes in connect function

### PR DESCRIPTION
**Pull request by isi-adas**
---Added retry in connect call to deal with sudden network issues

### CODE REVIEW

File: pike/model.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
 import time
 import operator
 import contextlib
+import polling2
```
Comment (masenf) [inline pike/model.py:56]: i think you need to add `polling2` to the `install_requires` in `setup.cfg`  if you're going to import it here.

**Review by masenf — CHANGES_REQUESTED** [review]
do you have plans to forward port this change?

**Comment by isi-adas** [comment]
> do you have plans to forward port this change?

I think there is no issues with forward porting this change. I will unless any concerns.

**Review by jtmoon79 — COMMENTED** [review]
Hi Arnab! 😄 

nitpick: Why not use the same variable name as the `polling`? e.g. `retry_limit` should be `max_tries`.  Seems silly that the same variable should be given such different names.

Otherwise, LGTM

**Comment by masenf** [comment]
That's a great call out, thanks @jtmoon79 and good to see you 👋.

I think the variable names should not be arbitrarily different.

**Comment by isi-adas** [comment]
Hey @jtmoon79 how is it going...Thanks for the callout. I will update.

**Review by masenf — APPROVED** [review]

**Pull request merged**